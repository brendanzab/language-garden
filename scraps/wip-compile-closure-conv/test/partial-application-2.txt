let add-captured : Int -> Int :=
  let x : Int := 5;
  let add-x : Int -> Int :=
    fun (y : Int) => x + y;

  add-x;

add-captured 3 + 4
