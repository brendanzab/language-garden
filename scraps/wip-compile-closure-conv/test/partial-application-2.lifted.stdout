def add-x0↑ (env1 : (Int,)) (y2 : Int) := #add env1.0 y2;
let add-captured3 : Int -> Int := let x0 : Int := 5; clos(add-x0↑, (x0,));
#add (add-captured3 3) 4
